orix/plot/unit_cell_plot.py

```diff
@@ -0,0 +1,199 @@
+        Keyword arguments passed to
+        :class:`matplotlib.patches.FancyArrowPatch`, for example "arrowstyle".
+
+    Returns
+    -------
+    fig : matplotlib.figure.Figure
+        The plotted figure.
+    """
+    # requires active rotation of the lattice in the sample reference frame
+    rotation = ~rotation
+
+    crystal_reference_frame_axes_position = (
+        crystal_reference_frame_axes_position.lower()
+    )
+    if not crystal_reference_frame_axes_position in ("origin", "center"):
+        raise ValueError(
+            'Crystal_reference_frame_axes_position must be either "origin" or "center".'
+        )
+    # TODO: More than only cubic
+    # introduce some basic non-cubic cell functionality
+    if structure is None:
+        structure = Structure(lattice=Lattice(2, 2, 2, 90, 90, 90))
+
+    assert isinstance(
+        structure, Structure
+    ), "Structure must be diffpy.structure.Structure."
+    lattice = structure.lattice
+    lattice_vectors = lattice.base
+
+    verts = _calculate_basic_unit_cell_vertices(lattice_vectors)
+    edges = _calculate_basic_unit_cell_edges(verts, lattice_vectors)
+    edges_rotated = rotation * Vector3d(edges)
+
+    fig, ax = plt.subplots(subplot_kw=dict(projection="3d"))
+    ax.axis("off")
+
+    xmax, ymax, zmax = np.max(np.abs(verts), axis=0)
+    lim = max(xmax, ymax, zmax)
+
+    pad = 1.5
+    axlim = pad * lim
+    ax.set_xlim(-axlim, axlim)
+    ax.set_ylim(-axlim, axlim)
+    ax.set_zlim(-axlim, axlim)
+    ax.set_box_aspect((1, 1, 1))  # equal aspect
+    ax.margins(0, 0, 0)
+
+    # default projection to +x -> east/right, +y -> north/upwards, +z out-of-page
+    ax.azim = -90
+    ax.elev = 90 - 1e-6  # add small offset to avoid arrow3D projection errors
+
+    # axes colors from ParaView and TomViz
+    colors = ("tab:red", "yellow", "tab:green")
+    labels = ("x", "y", "z")
+
+    arrow_kwargs.setdefault("mutation_scale", 20)
+    arrow_kwargs.setdefault("arrowstyle", "-")
+    arrow_kwargs.setdefault("linewidth", 2)
+
+    # add lab reference frame axes and labels
+    for i in range(3):
+        data = np.full((3, 2), -1.4 * lim)  # less padding than axlim
+        data[i, 1] += axes_length
+        label = labels[i]
+        arrow = Arrow3D(
+            *data,
+            color=colors[i],
+            label=f"Sample reference axes {label}",
+            **arrow_kwargs,
+        )
+        ax.add_artist(arrow)
+        # _s for sample
+        ax.text3D(
+            *data[:, 1], f"${label}_s$", label=f"Sample reference axes label {label}"
+        )
+
+    for i, (v1, v2) in enumerate(edges_rotated.data):
+        ax.plot3D(*zip(v1, v2), c=c, label=f"Lattice edge {i}")
+
+    # add crystal reference frame axes and labels
+    v_ref_ax = rotation * Vector3d(np.eye(3))
+    if crystal_reference_frame_axes_position == "origin":  # cell origin
+        crys_ref_ax_origin = Vector3d(verts[0])
+    else:
+        crys_ref_ax_origin = Vector3d(np.zeros(3))
```

Comment:
There's a method for this!

```suggestion
        crys_ref_ax_origin = Vector3d.zero(1)
```

Reply:
Thanks, didn't know this!